crates/wasmtime/src/module.rs

```diff
@@ -1042,6 +1042,34 @@ impl Module {
         self.inner.memory_images()?;
         Ok(())
     }
+
+    /// Get the map from `.text` section offsets to Wasm binary offsets for this
+    /// module.
+    ///
+    /// Each entry is a (`.text` section offset, Wasm binary offset) pair.
+    ///
+    /// Entries are yielded in order of `.text` section offset.
+    pub fn address_map<'a>(&'a self) -> impl Iterator<Item = (u32, Option<u32>)> + 'a {
+        wasmtime_environ::iterate_address_map(self.code_object().code_memory().address_map_data())
+            .expect("should have a valid address map section")
+            .map(|(offset, file_pos)| (offset, file_pos.file_offset()))
+    }
+
+    /// Get this module's code object's `.text` section, containing its compiled
+    /// executable code.
+    pub fn text(&self) -> &[u8] {
+        self.code_object().code_memory().text()
+    }
+
+    /// Get the locations of functions in this module's `.text` section.
+    ///
+    /// Each function's locartion is a (`.text` section offset, length) pair.
+    pub fn function_locations<'a>(&'a self) -> impl ExactSizeIterator<Item = (u32, u32)> + 'a {
```

Comment:
One thing I might recommend for this signature is that this would be a good place to convert from `u32` to `usize` since these are indexes into `self.text()`. Do you think it'd be worth returning `&[u8]` slices here though?

Reply:
The function slices themselves? These aren't actually that useful because everything is always text section relative offset wise, so I won't really ever be using the individual slices except when I slice them out myself and remember what offset the slices begin at.